src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverterFactoryHelpers.cs

```diff
@@ -254,13 +254,21 @@ private static Type GetImmutableDictionaryConstructingType(Type type)
 
         public static bool IsNonGenericStackOrQueue(this Type type)
         {
-            Type? typeOfStack = Type.GetType("System.Collections.Stack, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089");
-            Type? typeOfQueue = Type.GetType("System.Collections.Queue, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089");
+            Type? stackType = GetTypeIfExists("System.Collections.Stack, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089");
```

Comment:
Should this be referenced via `System.Collections.NonGeneric` so that it works even without the legacy `mscorlib` forwarder that is likely to be trimmed?

Reply:
This package targets netstandard2.0. Does the System.Collections.NonGeneric assembly exist in the GAC on downlevel platforms? I see that it's available in 4.8, but I don't have a 4.6.1 box to check on. Nor do I know how Mono / Xamarin / Unity / etc. handle this.

Reply:
This suggest we need to ifdefs this out. One path that works well downlevel and that is unfriendly to linking, and second path  for netcoreapp that works well for linking.